duolingo.py

```diff
@@ -118,8 +132,11 @@ def buy_item(self, item_name, abbr):
         returns a text like: {"streak_freeze":"2017-01-10 02:39:59.594327"}
         """
 
-        if request.status_code == 400 and request.json()['error'] == 'ALREADY_HAVE_STORE_ITEM':
-            raise AlreadyHaveStoreItemException('Already equipped with ' + item_name + '.')
+        if request.status_code == 400:
+            if request.json()["error"] == "ALREADY_HAVE_STORE_ITEM":
```

Comment:
might it happen that "error" key is not in the response?

Also, maybe it's better to do something like this, because string comparison with unofficial API is not that reliable, and would be helpful to see the exact error for an unhandled case.
```python
if request.status_code == 400:
  if request.json()["error"] == "ALREADY_HAVE_STORE_ITEM":
    raise AlreadyHaveStoreItemException("Already equipped with {}.".format(item_name))
  elif request.json()["error"] == "INSUFFICIENT_FUNDS":
    raise InsufficientFundsException("Insufficient funds to purchase {}.".format(item_name))
  else:
    raise DuolingoException("Unhandled exception: %s", request.json()["error"])
```

Reply:
Yeah, that sounds good. I've swapped the indexing for .get() so it won't raise KeyError in that case, and added handling for the unhandled case